### PR TITLE
expose metrics endpoint from within kpng

### PR DESCRIPTION
**Pull request by mattfenwick**
# What

Sets up prometheus server in kpng that can be scraped by prometheus master.

Sets up Grafana with prometheus as a data source.

Sets up a couple of default dashboards (for kube api server and nginx ingress controller).

Sets up an nginx ingress controller , so that Grafana can be accessed through ingress.

For some reason, the existing `--exposeMetrics` flag DOES NOT work => there seem to be problems with flag setup in the kpng cli.  Therefore, this adds a new function to set up the metrics server.

# How to use

See new hack/metrics/README.md

### CODE REVIEW

File: cmd/kpng/main.go [hunk 1]
```diff
@@ -44,10 +44,16 @@ var (
 
 // main starts the kpng program by running the command sent by the user.  This is the entry point to kpng!
 func main() {
+	metricsPort := "9099"
```
Comment (jayunit100) [inline cmd/kpng/main.go:47]: can u add a log statement in a follow on pr at some point saying metrics are on that port

**Review by jayunit100 — APPROVED** [review]

File: cmd/kpng/main.go (hunk 1 above)
Comment (mattfenwick) [reply cmd/kpng/main.go:47]: ` I0407 00:32:48.665494       1 metrics.go:38] Starting metrics server at 0.0.0.0:9099  ` already shows up in the logs, is that good enough?

Comment (mattfenwick) [reply cmd/kpng/main.go:47]: In context:

```
kubectl logs -f -n kube-system kpng-9dzjs kpng
I0407 00:32:48.665334       1 main.go:78] exporting metrics to: 0.0.0.0:9099 
I0407 00:32:48.665494       1 metrics.go:38] Starting metrics server at 0.0.0.0:9099
...
```

**Pull request merged**